plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/provider/SolidFireHostListener.java

```diff
@@ -78,9 +78,10 @@ public boolean hostAdded(long hostId) {
             return false;
         }
 
-        SolidFireUtil.hostAddedToCluster(hostId, host.getClusterId(), SolidFireUtil.PROVIDER_NAME,
-                clusterDao, hostDao, storagePoolDao, storagePoolDetailsDao);
-
+        if (host.getClusterId() != null) {
+            SolidFireUtil.hostAddedToCluster(hostId, host.getClusterId(), SolidFireUtil.PROVIDER_NAME,
+                    clusterDao, hostDao, storagePoolDao, storagePoolDetailsDao);
+        }
```

Comment:
@Pearl1594 _handleVMware()_ call may not be required when `host.getClusterId()` is null.

https://github.com/apache/cloudstack/blob/2c111715178583656dcc9d576906a92d101ef897/plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/provider/SolidFireHostListener.java#L161

https://github.com/apache/cloudstack/blob/2c111715178583656dcc9d576906a92d101ef897/plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/provider/SolidFireSharedHostListener.java#L146

Possible to return when cluster id is null ?

```
if (host.getClusterId() == null) {
   // add log or throw exception
    return false;
 }
```